src/extractor/guidance/segregated_intersection_classification.cpp

```diff
@@ -0,0 +1,244 @@
+#include "extractor/guidance/segregated_intersection_classification.hpp"
+#include "extractor/guidance/coordinate_extractor.hpp"
+#include "extractor/node_based_graph_factory.hpp"
+
+#include "util/coordinate_calculation.hpp"
+#include "util/name_table.hpp"
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+struct EdgeInfo
+{
+    NodeID node;
+
+    util::StringView name;
+
+    // 0 - outgoing (forward), 1 - incoming (reverse), 2 - both outgoing and incoming
+    int direction;
+
+    ClassData road_class;
+
+    RoadPriorityClass::Enum road_priority_class;
+
+    struct LessName
+    {
+        bool operator()(EdgeInfo const &e1, EdgeInfo const &e2) const { return e1.name < e2.name; }
+    };
+};
+
+bool IsSegregated(std::vector<EdgeInfo> v1,
+                  std::vector<EdgeInfo> v2,
```

Comment:
Not sure why we copy vectors here

Reply:
I think its because they are being sorted. Looks like the call site passes in the vectors by rvalue, so this should not be a problem.